### PR TITLE
Implement incremental directory sync

### PR DESCRIPTION
**Pull request by leondutoit**
* upload and download sync with the new `--upload-sync` and `--download-sync` options
* default behaviour: replace updated files, delete missing files, no caching (all configurable)
* depends on https://github.com/unioslo/tsd-file-api/pull/51
* bump version to 3.1.0

### CODE REVIEW

File: tsdapiclient/guide.py [hunk 1]
```diff
@@ -116,3 +116,30 @@
...
 
 """
+
+sync = f"""
+To incrementally synchronise directories:
+
+    tacl p11 --upload-sync mydir
+    tacl p11 --download-sync mydir
+
+By default, files that are present in the source, but missing
```
Comment (haatveit) [inline tsdapiclient/guide.py:126]:
```suggestion
By default, files that are present in the target, but missing
```

File: tsdapiclient/guide.py [hunk 2]
```diff
@@ -116,3 +116,30 @@
...
+in the target are deleted. Furthermore, if there are files in
```
Comment (haatveit) [inline tsdapiclient/guide.py:127]:
```suggestion
in the source are deleted. Furthermore, if there are files in
```

**Review by haatveit — APPROVED** [review]
Very nice!

File: tsdapiclient/guide.py (hunk 1 above)
Comment (leondutoit) [reply tsdapiclient/guide.py]: fixed 5202f0c

File: tsdapiclient/guide.py (hunk 2 above)
Comment (leondutoit) [reply tsdapiclient/guide.py]: fixed in 5202f0c

**Pull request merged**